extensions/dacpac/src/wizard/pages/deployConfigPage.ts

```diff
@@ -184,6 +158,30 @@ export class DeployConfigPage extends DacFxConfigPage {
 		}
 		let values = await this.getDatabaseValues();
 
+		/*
+		Check if databases exist for the selected server.
+		*/
+		if (values.length === 0 || !this.model.databaseExists) {
```

Comment:
why is this.model.databaseExists needed? I thought the error you were trying to fix was that  (<azdata.CategoryValue>this.databaseDropdown.value).name was undefined, which would be fixed by moving ```this.databaseDropdown.updateProperties({values: values})``` on line 190 up to line 160

Reply:
Sorry, I think I misunderstood, I thought I had to add another check on dacFxConfigPage to check if the server had any databases or not so that's why I created the this.model.databaseExists variable. I will remove it.

I moved line 190 before the values check and made the onDidClick appear only in the createRadioButtons function, and  there are no errors, but when I select a server with a database it keeps loading and only when I toggle between buttons it  displays the database name and dropdown, as discussed in the earlier comment. This issue was not appearing in the current commit that I had just pushed, in that the input components would change accordingly. Do you know how to debug this?

Reply:
I checked the code again and it seems that because of abstracting the onDidClick method for each radio button and calling the method that abstracts it prevents the loading issue. Do you think that would be an optimal solution or is there another way to debug it?

Reply:
checking for values.length === 0 is enough to check if the selected server has any databases. I think the error isn't showing up in the current changes because the onDidClick() handlers get set, but updateUpgradeRadioButton() doesn't get called. 

To debug, in vscode you can go to the Run viewlet on the left, then select "Attach to Extension Host" from the dropdown and then you'll be able to attach to the ADS and hit breakpoints in the dacpac extension. To start ADS, run ./sql.bat in the scripts folder. To see console errors, open up developer tools in ADS and go to the console tab.

Reply:
with the current changes, I was able to get into weird states like this where the inputbox doesn't get switched for the database dropdown when Upgrade Existing is selected 
![image](https://user-images.githubusercontent.com/31145923/97647281-13907d00-1a0f-11eb-97e3-04f6a3c20b4c.png)

and an empty dropdown and disabled Upgrade Existing
![image](https://user-images.githubusercontent.com/31145923/97647324-2d31c480-1a0f-11eb-873e-51239e544f18.png)

Reply:
I've made some changes as suggested, I've gotten rid of the excess onDidClick methods, and also added some checks for the number of pages that were added, as there were instances where multiple pages were added due to clicking on the radio buttons continuously. I suspect that could be one of the reasons why the database loader keeps loading. There are no errors in the console, but there are still instances where the loader continues to load, could you please test and find out any alternatives for this?

Reply:
ah good catch on the multiple pages being added! 

For the loading spinner not going away, I was able to get that to go away by moving `this.databaseLoader.loading = false;` up to line 170. I'm not sure why that fixed it, but probably has something to do with it getting set to false before the databaseDropdownComponent is added back

Reply:
There was an small bug occurring when I moved from step 3 in the wizard to step 2, as shown below.
![newerror](https://user-images.githubusercontent.com/29865459/97760252-f1a30300-1abf-11eb-870a-f4c91621510d.png)

It was pointing to the new radio button with the dropdown component. I was able to fix this by making a small check to see if the new radio button is selected then to set the newRadioButton.checked to false. We would want the upgrade radio button to be selected in this case, but when I set the upgradeRadioButton.checked to true, it doesn't get selected, so it has been modified to this:
![newerrors](https://user-images.githubusercontent.com/29865459/97760346-2f079080-1ac0-11eb-8079-72269cf8baa0.png)
Other than that, the other issues seem to be resolved after moving the databaseLoader.loading variable, and the remaining pages in the wizard seem to work properly. Could you please test this to see if everything is in place?

Reply:
I'm running into this state when switching to a server with no databases. The New Database radio button should be selected since the upgrade button is disabled.
![image](https://user-images.githubusercontent.com/31145923/97762287-439a5780-1ac5-11eb-8d8d-ca08e94428ff.png)

After this is fixed, this should be good to check in!